shared/chat/conversation/input/container.js

```diff
@@ -103,9 +104,12 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
   onJoinChannel: (selectedConversation: Constants.ConversationIDKey) => {
     dispatch(ChatGen.createJoinConversation({conversationIDKey: selectedConversation}))
   },
-  onLeaveChannel: (selectedConversation: Constants.ConversationIDKey) => {
+  onLeaveChannel: (selectedConversation: Constants.ConversationIDKey, teamname: string) => {
     dispatch(ChatGen.createLeaveConversation({conversationIDKey: selectedConversation}))
-    dispatch(ChatGen.createSelectConversation({conversationIDKey: null}))
+    dispatch(navigateUp())
+    if (ownProps.fromManageChannels) {
+      dispatch(navigateAppend([{props: {teamname}, selected: 'manageChannels'}]))
```

Comment:
The routing behavior logic was a little confusing here; if anyone has a better suggestion let me know please!